WordPress/Classes/ViewRelated/Post/PostSharingController.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import SVProgressHUD
+
+@objc class PostSharingController : NSObject {
+    
+    func shareController(title:String?, summary:String?, tags:String?, link:String?) -> UIActivityViewController {
+        var activityItems = [AnyObject]()
+        let postDictionary = NSMutableDictionary()
+        
+        if let str = title {
+            postDictionary["title"] = str
+        }
+        if let str = summary {
+            postDictionary["summary"] = str
+        }
+        if let str = tags {
+            postDictionary["tags"] = str
+        }
+        
+        activityItems.append(postDictionary)
+        if let urlPath = link, url = NSURL(string: urlPath) {
+            activityItems.append(url)
+        }
+        
+        let activities = WPActivityDefaults.defaultActivities() as! [UIActivity]
+        let controller = UIActivityViewController(activityItems: activityItems, applicationActivities: activities)
+        if let str = title {
+            controller.setValue(str, forKey:"subject")
+        }
+        controller.completionWithItemsHandler = {
+            (activityType:String?, completed:Bool, items: [AnyObject]?, error: NSError?) in
+            
+            if completed {
+                WPActivityDefaults.trackActivityType(activityType)
+            }
+        }
+        
+        return controller
+    }
+    
+    func sharePost(title: String, summary: String, tags: String, link: String, fromView anchorView:UIView, inViewController viewController:UIViewController) {
+        let controller = shareController(
+            title,
+            summary: summary,
+            tags: tags,
+            link: link)
+        
+        if !UIDevice.isPad() {
+            viewController.presentViewController(controller, animated: true, completion: nil)
+            return
+        }
+        
+        // Silly iPad popover rules.
+        controller.modalPresentationStyle = .Popover
+        viewController.presentViewController(controller, animated: true, completion: nil)
+        if let presentationController = controller.popoverPresentationController {
+            presentationController.permittedArrowDirections = .Unknown
+            presentationController.sourceView = anchorView
+            presentationController.sourceRect = anchorView.bounds
+        }
+    }
+    
+    func sharePost(post: Post, fromView anchorView:UIView, inViewController viewController:UIViewController) {
```

Comment:
I started aiming for that, but found a few technical issues making it a bit harder than I would have liked for the scope of this PR.  More specifically: the prototype didn't like having methods named as some of the properties from the classes implementing the protocol.

Regardless of that, though, I was hoping to review this particular issue separately because I feel this could have a better solution.  If not, I'll make sure to submit a separate PR with a protocol to take care of this (but I'd still like to abstract that "issue" from this specific PR).